Fixed #25901

divyajyothi5321 opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#25901: Mobile design issues in Minicart Dropdown Dialog

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Add item to cart.
2. Now open minicart dropdown dialog in mobile Portrait and Landscape view.
3. Change the qty and click out of the qty box so that update button will be displayed.
4. After this fix is applied the issue will be fixed as shown in the below screenshots.

![update-cart-expected-result](https://user-images.githubusercontent.com/54176640/70155125-af1e4e00-16d7-11ea-9cd9-ecc0c1bf6aa0.png)


![edit-delete-expected](https://user-images.githubusercontent.com/54176640/70155124-ae85b780-16d7-11ea-8bfe-db4304e80be9.png)

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

divyajyothi5321 committed:
Fixed #25901

divyajyothi5321 committed:
Added necessary Indentations

krzksz reviewed (changes requested):
Please use `.media-width` mixin to define media queries and `@indent__` variables.

divyajyothi5321 committed:
Added the suggested changes

krzksz commented on app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -431,6 +431,33 @@
 }
 
+.media-width(@extremum, @break) when (@extremum = 'max') and (@break = @screen__xs) {
+    .minicart-items .item-qty {
```
I believe this can be merged with the one at line 412.

krzksz commented on app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -431,6 +431,33 @@
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum = 'max') and (@break = @screen__m) {
```
This can be merged with line 418.

krzksz commented on app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -431,6 +431,33 @@
+
+@media only screen 
+and (min-device-width: @screen__xs) 
+and (max-device-width: @screen__m) {
```
This has to use two separate `.media-width`, one for minimum `@screen__xs` and one for minimum `@screen__m` that sets those values back.

krzksz replied:
Maybe what I wrote is not clear enough here. I would like you to split this media query into 2 separate ones so we can use `.media-width` is such a way:
```less
.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__xs) {
  // your styles with margin and padding.
}

.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__m) {
  // styles reverting margin and padding for bigger screens.
}
```

krzksz replied:
This one is still not addressed.

divyajyothi5321 committed:
Added the suggested changes

krzksz commented on app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -429,6 +432,25 @@
             float: right;
         }
     }
+    .minicart-items .product-item-details .details-qty {
```
Please add an empty line before.

krzksz commented on app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -413,6 +413,9 @@
     .minicart-wrapper .block-minicart {
         width: 290px;
     }
+    .minicart-items .item-qty {
```
Please add an empty line before.

krzksz reviewed (changes requested):
Please make the changes I requested.

divyajyothi5321 committed:
Changed in the latest commit

divyajyothi5321 committed:
Added New Changes

We have Fixed in different way in order to work in all screens by using media-width.

magento-engcom-team commented:
Hi @krzksz, thank you for the review.
[ENGCOM-6438](https://jira.corp.magento.com/browse/ENGCOM-6438) has been created to process this Pull Request
:eight_spoked_asterisk: @krzksz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

magento-engcom-team commented:
@divyajyothi5321 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

krzksz commented:
Purely visual change, no tests required.

engcom-Alfa commented:
Hi @divyajyothi5321 

Unfortunately, we are not able to reproduce this [issue](https://github.com/magento/magento2/issues/25901) on the fresh 2.4-develop instance

**Actual Result:**
Portrait
![before](https://user-images.githubusercontent.com/51679138/70702207-6289d280-1cd6-11ea-83f4-ee3a948f405c.png)

Landscape
![before2](https://user-images.githubusercontent.com/51679138/70702225-6cabd100-1cd6-11ea-993c-42035ebf5085.png)

Looks like this issue was already fixed in PR [#25206](https://github.com/magento/magento2/pull/25206)

@divyajyothi5321 Could you take a look?

Thanks!
